src/Mustard/Extension/Geant4X/Geometry/HelicalBox.c++

```diff
@@ -111,15 +116,80 @@ HelicalBox::HelicalBox(std::string name,
         c(i, 1) = AuxillaryPoint(u[i], 1);
     }
 
+    // compute end position and normal
+    const auto Helix{
+        [&](const auto& u) -> G4Point3D {
+            const auto u1{u + phi0};
+            return {radius * std::cos(u1),
+                    radius * std::sin(u1),
+                    u1 * tanAR - zOffset};
+        }};
+    const auto EndFaceNormal{
+        [&](const auto& u) -> G4ThreeVector {
+            const auto u1{u + phi0};
+            return {-radius * std::sin(u1),
+                    radius * std::cos(u1),
+                    tanAR};
+        }};
+    fFrontEndPosition = Helix(0);
+    fFrontEndNormal = EndFaceNormal(0).unit();
+    fBackEndPosition = Helix(phiTotal);
+    fBackEndNormal = EndFaceNormal(phiTotal).unit();
+
+    auto CreateParallelPoints{[&](const Eigen::MatrixX<G4Point3D>& grid,
+                                  int index,
+                                  double endZ,
+                                  const G4ThreeVector& endNormal) -> std::vector<G4ThreeVector> {
+        std::vector<G4ThreeVector> projectedPoints;
+
+        for (int j{}; j < 4; j++) {
+            const G4Point3D& p{grid(index, j)};
+            double t{(endZ - p.z()) / endNormal.z()};
+            if (endZ < 0 and frontParallel) {
+                double t0{(x(1, j).z() - p.z()) / endNormal.z()};
+                if (std::abs(t) > std::abs(t0)) {
+                    Mustard::Throw<std::runtime_error>("the front end can not be parallel to the z-axis!");
```

Comment:
Not parallel to z-axis but perpendicular to it.

Also we should consider the clarity of `frontParallel`/`backParallel`, which still seems confusing. We can seek help from Deepseek for a maybe more resonable naming.